packages/ddex/ingester/crawler/crawler.go

```diff
@@ -114,245 +101,185 @@ func (c *Crawler) getLastPolledStr() (string, error) {
-	if err != mongo.ErrNoDocuments {
-		c.Logger.Error("Error querying Mongo to check for existing remotePath", "remotePath", delivery.RemotePath, "error", err)
-		return err
-	}
+	for _, item := range res.Contents {
+		if strings.HasSuffix(*item.Key, ".xml") {
+			// If we have an XML file, insert it into the Mongo releases or batches collection
+			if err = c.upsertXML(*item.Key, *item.LastModified); err != nil {
+				return
+			}
+		} else if strings.HasSuffix(*item.Key, ".zip") {
+			// If we have a ZIP file, re-upload it as an unzipped folder without messing up the cursor
 
-	c.Logger.Info("Processing new delivery", "remotePath", delivery.RemotePath)
+			// Download the ZIP file to a temp dir
+			remotePath := fmt.Sprintf("s3://%s/%s", c.Bucket, *item.Key)
+			var deliveryLocalPath string
+			deliveryLocalPath, err = os.MkdirTemp("", *item.Key+"-*")
+			if err != nil {
+				err = fmt.Errorf("error creating temp directory: %w", err)
+				return
+			}
+			defer func() {
+				if err := os.RemoveAll(deliveryLocalPath); err != nil {
+					c.Logger.Error("Error removing temp directory", "deliveryLocalPath", deliveryLocalPath, "error", err)
+				}
+			}()
+			zipFilePath, cleanup := c.downloadFileFromS3Raw(remotePath)
+			defer cleanup()
+			if zipFilePath == "" {
+				err = fmt.Errorf("failed to download ZIP file from S3 path: %s", remotePath)
+				return
+			}
 
-	// Create temp directory for the delivery's contents (either from ZIP file or folder in S3)
-	deliveryLocalPath, err := os.MkdirTemp("", "delivery")
-	if err != nil {
-		c.Logger.Error("Error creating temp directory", "error", err)
-		return err
-	}
-	defer func() {
-		if err := os.RemoveAll(deliveryLocalPath); err != nil {
-			c.Logger.Error("Error removing temp directory", "deliveryLocalPath", deliveryLocalPath, "error", err)
-		}
-	}()
+			// Unzip
+			if err = unzip(zipFilePath, deliveryLocalPath); err != nil {
+				err = fmt.Errorf("error unzipping file: %w", err)
+				return
+			}
 
-	// Download folder from S3, or download+extract ZIP file from S3
-	if delivery.IsFolder {
-		err := c.downloadFolderFromS3Raw(delivery.RemotePath, deliveryLocalPath)
-		if err != nil {
-			c.Logger.Error("Failed to download folder from S3", "path", delivery.RemotePath, "error", err)
-			return err
-		}
-	} else {
-		zipFilePath, cleanup := c.downloadFileFromS3Raw(delivery.RemotePath)
-		defer cleanup()
-		if zipFilePath == "" {
-			c.Logger.Error("Failed to download ZIP file from S3", "path", delivery.RemotePath)
-			return err
-		}
+			// Re-upload the unzipped contents
+			var entries []fs.DirEntry
+			entries, err = fs.ReadDir(os.DirFS(deliveryLocalPath), ".")
+			if err != nil {
+				err = fmt.Errorf("failed to read directory '%s': %w", deliveryLocalPath, err)
+				return
+			}
 
-		if err := unzip(zipFilePath, deliveryLocalPath); err != nil {
-			c.Logger.Error("Error unzipping file", "error", err)
-			return err
-		}
-	}
+			for _, entry := range entries {
+				if entry.Name() == ".DS_Store" {
```

Comment:
would pull this out into a const of skippable files

Reply:
changed